### PR TITLE
[JS/WebGPU] Expand operator fixes

### PR DESCRIPTION
**Pull request by hariharans29**
### Description
1) The `cacheHint` for `Expand` is set to be a function of the input tensors' shapes. This is a problem because the shader is a function of the first input's shape and the second input's contents. Without this fix, we could be using a cached `ProgramInfo` incorrectly in places where we should be using a different `ProgramInfo`. A test is checked-in for this.

2) Support the first input tensor being 1-D. The  `const inputShape = ${input.indices(...inputShape)};` returns an `u32` for a 1-D tensor which means that we can't be using logic like `inputShape[i]` in that case as it is not an `array`. A similar (but not exactly the same) issue was fixed in https://github.com/microsoft/onnxruntime/pull/16829/ but after the indices helper refactoring a similar issue was re-introduced back. A fix is proposed and a test is added for this.

### Motivation and Context
Fix some bugs in `Expand` WebGPU

### CODE REVIEW

File: js/web/test/data/ops/expand.jsonc [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            "data": [1, 1, 1, 1, 1, 1],
+            "dims": [1, 6],
+            "type": "float32"
+          }
+        ]
+      }
+    ]
+  },
+  {
+    "name": "Expand 1D - float32",
```
Comment (hariharans29) [inline js/web/test/data/ops/expand.jsonc:61]: Test for second bug fix

File: js/web/test/data/ops/expand.jsonc [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            "data": [1, 1, 1, 1],
+            "dims": [1, 4],
+            "type": "float32"
+          }
+        ]
+      }
+    ]
+  },
+  {
+    "name": "Expand 2 - float32",
```
Comment (hariharans29) [inline js/web/test/data/ops/expand.jsonc:32]: Test for first bug fix - input tensors are of shape shapes but the second tensor's contents are different.

**Review by fs-eire — APPROVED** [review]

File: js/web/lib/wasm/jsep/webgpu/ops/expand.ts [hunk 3]
```diff
@@ -66,7 +66,6 @@ const createExpandProgramInfo = (metadata: ProgramMetadata, inputs: readonly Ten
     var inputIndices: ${input.type.indices};
     for (var i = 0; i < ${inputShape.length}; i++) {
```
[the remark this answers is not in the document]
Comment (hariharans29) [reply js/web/lib/wasm/jsep/webgpu/ops/expand.ts:67]: Used option B instead of current option A

**Review by satyajandhyala — APPROVED** [review]

File: js/web/lib/wasm/jsep/webgpu/ops/expand.ts [hunk 4]
```diff
@@ -86,7 +85,8 @@ const createExpandProgramInfo = (metadata: ProgramMetadata, inputs: readonly Ten
 
 export const expand = (context: ComputeContext): void => {
   validateInputs(context.inputs);
-  const cacheHint = context.inputs.map(x => x.dims.toString()).join('_');
+  const outputShape = Array.from(context.inputs[1].getBigInt64Array(), Number);
```
Comment (satyajandhyala) [inline js/web/lib/wasm/jsep/webgpu/ops/expand.ts:88]: Does the cacheHint depend input shape also?

Comment (hariharans29) [reply js/web/lib/wasm/jsep/webgpu/ops/expand.ts:88]: Input shape info is already included by the upper layer code - https://github.com/microsoft/onnxruntime/blob/99349e58d7127ae8ce268c31b299aeb75a941c42/js/web/lib/wasm/jsep/backend-webgpu.ts#L24

**Review by satyajandhyala — APPROVED** [review]

**Pull request merged**